internal/controllers/clustertemplate_controller.go

```diff
@@ -284,15 +297,22 @@ func validateName(c client.Client, name, version, metadataName, namespace string
 
 // validateTemplateID return true if the templateID is a valid uuid, false otherwise
 // If the templateID does not exist, it is generated by this function using uuid
-func validateTemplateID(ctx context.Context, c client.Client, object *provisioningv1alpha1.ClusterTemplate) error {
+func validateTemplateID(object *provisioningv1alpha1.ClusterTemplate) error {
 	if object.Spec.TemplateID != "" {
 		_, err := uuid.Parse(object.Spec.TemplateID)
 		if err != nil {
 			return utils.NewInputError("failed to validate templateID, invalid UUID:%s", object.Spec.TemplateID)
 		}
-		return nil
 	}
+	return nil
+}
 
+// generateTemplateID generates a new templateId if it was not present
+// If the templateID does not exist, it is generated by this function using uuid
+func generateTemplateID(ctx context.Context, c client.Client, object *provisioningv1alpha1.ClusterTemplate) error {
```

Comment:
I think c.Get(line324-327) in this function could be removed

Reply:
Fixed